api/include/opentelemetry/logs/log_record.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <map>
+#include <unordered_map>
+#include "opentelemetry/common/key_value_iterable_view.h"
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_flags.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace logs
+{
+
+/* Note: using a class enum here won't allow enum values to be compared to integers, i.e. only other
+ * Severity enums (need an explicit cast)
+ * Follows the Google standard for naming:
+ * https://google.github.io/styleguide/cppguide.html#Enumerator_Names
+ */
+enum class Severity : uint8_t
+{
+  kNone   = 0,  // default Severity; added, not part of Log Data Model
+  kTrace  = 1,
+  kTrace2 = 2,
+  kTrace3 = 3,
+  kTrace4 = 4,
+  kDebug  = 5,
+  kDebug2 = 6,
+  kDebug3 = 7,
+  kDebug4 = 8,
+  kInfo   = 9,
+  kInfo2  = 10,
+  kInfo3  = 11,
+  kInfo4  = 12,
+  kWarn   = 13,
+  kWarn2  = 14,
+  kWarn3  = 15,
+  kWarn4  = 16,
+  kError  = 17,
+  kError2 = 18,
+  kError3 = 19,
+  kError4 = 20,
+  kFatal  = 21,
+  kFatal2 = 22,
+  kFatal3 = 23,
+  kFatal4 = 24
+};
+
+/* _nullKV is defined as a private variable that allows "resource" and
+    "attributes" fields to be instantiated using it as the default value */
+static common::KeyValueIterableView<std::map<nostd::string_view, nostd::string_view>> _nullKV =
+    common::KeyValueIterableView<std::map<nostd::string_view, nostd::string_view>>{{}};
+
+/**
+ * A default Event object to be passed in log statements,
+ * matching the 10 fields of the Log Data Model.
+ * (https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/logs/data-model.md#log-and-event-record-definition)
+ *
+ */
+struct LogRecord
+{
+  // default fields that will be set if the user doesn't specify them
+  core::SystemTimestamp timestamp;  // uint64 nanoseconds since Unix epoch
+  trace::TraceId trace_id;          // byte sequence
+  trace::SpanId span_id;            // byte sequence
+  trace::TraceFlags trace_flag;     // byte
+  Severity severity;  // Severity enum that combines severity_text and severity_number in the
+                      // LogDataModel (can separate in SDK)
+
+  // other fields that will not be set by default
+  nostd::string_view name;  // string
+  nostd::string_view body;  // currently a simple string, but should be changed "Any" type
+  common::KeyValueIterable &resource;    // key/value pair list
+  common::KeyValueIterable &attributes;  // key/value pair list
+
+  /* Default log record if user does not overwrite this.
+   *  TODO: find better data type for "body" field
+   *  Future enhancement: Potentially add other constructors to take default arguments from the user
+   **/
+  LogRecord() : resource(_nullKV), attributes(_nullKV)
+  {
+    timestamp = core::SystemTimestamp(std::chrono::system_clock::now());
```

Comment:
I think this needs to be moved to the SDK.
Consider adding a TODO comment here.